### DISCUSSION TITLE
Split nanodbc.cpp into multiple files?

### DISCUSSION THREAD

**Discussion by mloskot** [nanodbc/nanodbc#306] [Ideas]
How much are we attached to the idea of maintaining single `nanodbc.cpp`?

The obvious advantage is that it makes it easy to add `nanodbc.cpp` directly to a user's project.
The disadvantage that I see is the maintenance of **6.5 KLOC** in single file.

What's your thought on the splitting, @lexicalunit?
@jimhester, if I may call you out, does r-dbi rely on single `nanodbc.cpp`?

**Comment by jimhester** [comment]
We could use multiple files if needed, though having it in a single file does make it easier to include.

If maintenance is the primary concern one solution would be to split up the file but also provide the ability to construct an amalgamated version for distribution using something like https://bitbucket.org/erikedlund/amalgamate/src/master/.

**Reply by mloskot** [reply]
Yes, amalgamation is a good idea and if the splitting ever happens, then we should provide amalgamated distribution indeed.
